### ISSUE TITLE
[accessibility] iOS Voiceover cuts off Snackbar semantics on navigating to a page with Snackbar

### ISSUE THREAD

**Issue by ANPetru** [flutter/flutter#83782]
## Steps to Reproduce

1. Run `flutter create bug`.
2. Update `main.dart` file with `main.txt` 
[main.txt](https://github.com/flutter/flutter/files/6583528/main.txt)
3. Run app on Android
4. Activate talkback
5. Navigate to second page, notice talkback focuses snackbar
6. Run app on iOS
7. Activate Voiceover
8. Navigate to second page, notice voiceover focuses backbutton

**Expected results:** 
When opening a new page and displaying a snackbar, talkback and voiceover should focus it first.

**Actual results:** 
When opening a new page and displaying a snackbar, Talkback focuses the snackbar first. Voiceover focuses backbutton instead of snackbar.  

Both accessibility tools should focus same widget.

<details> 
<summary>flutter doctor -v</summary> 
	
```console
[✓] Flutter (Channel unknown, 1.22.6, on Linux, locale es_ES.utf8)
    • Flutter version 1.22.6 at /home/usuario/fvm/versions/1.22.6
    • Framework revision 9b2d32b605 (hace 4 meses), 2021-01-22 14:36:39 -0800
    • Engine revision 2f0af37152
    • Dart version 2.10.5

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /home/usuario/Android/Sdk
    • Platform android-30, build-tools 29.0.2
    • ANDROID_HOME = /home/usuario/Android/Sdk
    • Java binary at: /usr/local/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)
    • All Android licenses accepted.

[✓] Android Studio (version 3.3)
    • Android Studio at /usr/local/android-studio
    • Flutter plugin version 34.0.1
    • Dart plugin version 182.5215
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)

[✓] Connected device (1 available)
    • SM A520F (mobile) • 5210ef4843b93469 • android-arm64 • Android 8.0.0 (API 26)

• No issues found!
```

</details>

**Comment by ANPetru** [comment]
Hey, I upgraded to the lastest version of flutter, 2.2.1, and noticed it actually focuses snack bar first.

The problem is that the snack bar semantic gets randomly cut by the back button semantic.  

I made a video where you can see it clearly.
https://user-images.githubusercontent.com/36933718/121057348-775a0000-c7bf-11eb-9c15-541ab8b50cdd.mp4

In your iOS example it also gets cut, if you listen closely, It says "Test snack ba-Back button", while on Android it waits for the snack bar semantic to end: "Test snack bar. Second page. Back button"


<details>
  <summary>Minimal code sample</summary>


```import 'package:flutter/material.dart';

void main() {
  runApp(MaterialApp(
    title: 'Navigation Basics',
    home: FirstRoute(),
  ));
}

class FirstRoute extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('First Route'),
      ),
      body: Center(
        child: ElevatedButton(
          child: Text('Open route'),
          onPressed: () {
            Navigator.push(
              context,
              MaterialPageRoute(builder: (context) => SecondRoute()),
            );
          },
        ),
      ),
    );
  }
}

class SecondRoute extends StatelessWidget {

  @override
  Widget build(BuildContext context) {
    return Scaffold(
        appBar: AppBar(
          title: Text("Second Page"),
        ),
        body: Builder(builder: (context) {
          Future.delayed(
            Duration.zero,
            () => Scaffold.of(context).showSnackBar(
              SnackBar(
                content: Text(
                  "Test snack bar",
                  textAlign: TextAlign.center,
                ),
                duration: const Duration(seconds: 30),
              ),
            ),
          );
          return Container();
        }));
  }
}
```
</details>
<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 2.2.1, on macOS 11.2.3 20D91 darwin-arm, locale en-ES)
    • Flutter version 2.2.1 at /opt/flutter
    • Framework revision 02c026b03c (11 days ago), 2021-05-27 12:24:44 -0700
    • Engine revision 0fdb562ac8
    • Dart version 2.13.1

[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.


[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5, Build version 12E262
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] VS Code (version 1.56.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.23.0

[✓] Connected device (1 available)
    • Chrome (web) • chrome • web-javascript • Google Chrome 91.0.4472.77

! Doctor found issues in 1 category.
```
</details>


Thank you!

**Comment by TahaTesser** [comment]
Hi @ANPetru 
That's true, it does get cut off (Also reproduces on iOS 15)

### Used the following minimal code sample

<details> 
<summary>minimal code sample</summary> 

```dart
``` 

</details>


### Preview
https://user-images.githubusercontent.com/48603081/121136298-2806e480-c83e-11eb-959e-c2b6f8bb1e7e.mov

### Logs

|  `stable`  | `master` |
| :---: |:---: |
|  ❗  |   ❗  | 

Check `flutter doctor -v` outputs for each channel below

<details> 
<summary>flutter doctor -v</summary> 
	
```console

```

```console
```
		
</details>

✅ : No Issue ❗: Issue reproduced

**Comment by ANPetru** [comment]
Thanks!
